spec/radioactive_bunnies/deadletter_worker_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'spec_helper'
+require 'radioactive_bunnies'
+require 'support/workers/timeout_worker'
+JOB_TIMEOUT = 1
+class DeadletterDefaultWorker
+  include RadioactiveBunnies::Worker
+  from_queue 'deadletter.default',
+    prefetch: 20, durable: false, timeout_job_after: 5, threads: 1, append_env: true,
+    exchange: {type: :fanout, name: 'deadletter.exchange'}
+  def work(metadata, msg)
+    true
+  end
+end
+
+class DeadletterDefaultWorkerTwo
+  include RadioactiveBunnies::Worker
+  from_queue 'deadletter.default.two',
+    prefetch: 20, durable: false, timeout_job_after: 5, threads: 1, append_env: true,
+    exchange: {type: :fanout, name: nil}
+
+  def work(metadata, msg)
+    true
+  end
+end
+
+class DeadletterProducer
+  include RadioactiveBunnies::Worker
+  from_queue 'deadletter.producer',
+    prefetch: 20, timeout_job_after: JOB_TIMEOUT, threads: 2,
+    deadletter_workers: ['DeadletterDefaultWorker']
+  def work(metadata, msg)
+    sleep (JOB_TIMEOUT + 5)
+  end
+end
+
+class DeadletterProducerNonArray
+  include RadioactiveBunnies::Worker
+  from_queue 'deadletter.producer.nonarray',
+    prefetch: 20, timeout_job_after: 1, threads: 2,
+    deadletter_workers: 'DeadletterDefaultWorker'
+  def work(metadata, msg)
+    false
+  end
+end
+
+class DeadletterProducerBroken
+  include RadioactiveBunnies::Worker
+  from_queue 'deadletter.producer.nonarray',
+    prefetch: 20, timeout_job_after: 1, threads: 2,
+    deadletter_workers: %w(DeadletterDefaultWorker DeadletterDefaultWorkerTwo)
+  def work(metadata, msg)
+    false
+  end
+end
+
+describe RadioactiveBunnies::DeadletterWorker do
+
+  before(:all) do
+    @conn = MarchHare.connect
+    @ch = @conn.create_channel
+    @ctx = RadioactiveBunnies::Context.new
+    @ctx.log_with(Logger.new(STDOUT))
+    @ctx.run DeadletterProducer, DeadletterDefaultWorker, DeadletterProducerNonArray
+    ['deadletter.producer', 'deadletter.producer.nonarray'].each do |r_key|
+      @ch.default_exchange.publish("hello world", routing_key: r_key)
+    end
+    sleep 2
+  end
+
+  after(:all) do
+    @conn.close
+    @ctx.stop
+  end
+  context 'when a worker has at least one deadletter class' do
+
+    it 'notifies the deadletter worker that it will produce deadletters' do
+      expect(DeadletterDefaultWorker.deadletter_producers).to include(DeadletterProducer.name)
+    end
+    it 'notifies the deadletter worker when a single deadletter work is identified' do
```

Comment:
well we might have an issue because I'm a bit confused on your comment.  Can we elaborate?

If you are talking about include(DeadletterProducer.name) - this is just Class.name? Is that the confusing part?